app/js/components/manageApps/SingleAddon.jsx

```diff
@@ -0,0 +1,85 @@
+// /*
+//  * This Source Code Form is subject to the terms of the Mozilla Public License,
+//  * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+//  * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+//  * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+//  *
+//  * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+//  * graphic logo is a trademark of OpenMRS Inc.
+//  */
+import React from 'react';
+import { Link } from 'react-router';
+
+export default class SingleAddon extends React.Component{
+    constructor(props){
+        super(props);
+    }
+
+    render(){
+        const {
+            app,
+            key,
+            addonParam,
+            updatesVersion} = this.props
+
+        return(
+            <tr key={key}>
+                <td onClick={() => openPage(app.appDetails)}>
+                    <img
+                    src={app.appDetails &&
```

Comment:
Shouldn't this be indented as it lives inside the <img /> tag?

I am also a little curious as to  what happens when `app.appDetails.icon === null` or `app.appDetails.icons === undefined`, how do you capture that?

Reply:
Consideration for whether there is data within app is done by the parent components that call SingleAddon for example ManageApps and AddonList